src/bit_encoding/decode.rs

```diff
@@ -375,7 +375,10 @@ impl<'a, I: Iterator<Item = u8>> WitnessDecoder<'a, I> {
 
 impl<'a, I: Iterator<Item = u8>> WitnessIterator for WitnessDecoder<'a, I> {
     fn next(&mut self, ty: &types::Final) -> Result<Value, crate::Error> {
-        decode_value(ty, self.bits).map_err(From::from)
+        self.bits
+            .read_value(ty)
+            .map_err(Error::from)
+            .map_err(crate::Error::from)
```

Comment:
29a2250ed158a237204f8c06eb9ddf560b989f46: `self.bits.read_value(ty).map_err(crate::Error::from)` works directly because of `From<EarlyEndOfStreamError> for Error` in `lib.rs`